api/notd/manager.py

```diff
@@ -80,6 +80,54 @@ async def retrieve_ui_data(self, startDate: datetime.datetime, endDate: datetime
             sponsoredToken=self.get_sponsored_token(),
             transactionCount=transactionCount
         )
+    async def get_collection_stats(self, address: str, startDate: datetime.datetime, endDate: datetime.datetime) -> CollectionStatistics:
+        items = await self.retriever.list_token_metadatas(
+            fieldFilters=[
+                StringFieldFilter(fieldName=TokenTransfersTable.c.registryAddress.key, eq=address),
+            ],
+        )
+        itemCount = len(items)
+
+        holders = await self.retriever.list_token_transfers(
```

Comment:
this is very expensive - some collections will have tens or hundreds of thousands of transfers. can you find out long this api takes for the biggest collections e.g. BAYC